### PR TITLE
Adding RedHat support

### PR DESCRIPTION
**Pull request by phathocker**

### CODE REVIEW

**Comment by phathocker** [comment]
@dhoppe - What do you think?  Do I need to do anything more to support the RedHat based OSes?  My branch is working in my lab environment.

**Comment by nicolasbrechet** [comment]
I'd love to this this getting merged!  :+1: 
@dhoppe any chance of getting this merged ? Thanks

**Comment by dhoppe** [comment]
@phathocker Thanks for your contribution. Could you please do a rebase to solve the conflicts?

**Comment by phathocker** [comment]
@dhoppe - I merged master but I don't think your Travis tests are working.  Do you still plan to support ruby versions less than 2.2.5?

`ruby_dep-1.5.0 requires ruby version >= 2.2.5, which is incompatible with the
current version, ruby 2.1.7p400`

**Comment by phathocker** [comment]
I also have #18 with passing tests if you'd rather merge it.

File: metadata.json [hunk 1]
```diff
@@ -41,6 +49,9 @@
     }
   ],
   "tags": [
-    "fail2ban"
+    "fail2ban",
+    "debian",
+    "ubuntu",
+    "redhat"
```
Comment (dhoppe) [inline metadata.json]: Please remove lines 53 to 55, because the operating system might be bad for the sore at Puppet Forge.

**Comment by dhoppe** [comment]
@phathocker Please do a rebase to get the latest changes.

File: metadata.json [hunk 2]
```diff
@@ -45,6 +53,6 @@
     }
   ],
   "tags": [
-    "fail2ban"
+    "fail2ban",
```
Comment (dhoppe) [inline metadata.json]: @phathocker You need to remove the `,` or this will fail during `metadata-json-lint`.

**Comment by dhoppe** [comment]
@phathocker One last favour. Please use `scope['xyz']` instead of `scope.lookupvar('xyz')`. 
https://docs.puppet.com/puppet/latest/reference/lang_template_erb.html#scopevariable-or-scopelookupvarvariable

**Pull request merged**